test/test_utils/sagemaker.py

```diff
@@ -198,6 +194,39 @@ def install_sm_local_dependencies(framework, job_type, image, ec2_conn):
     ec2_conn.run(f"sudo {is_py3} pip install -r requirements.txt ", warn=True)
 
 
+def kill_background_processes_and_run_apt_get_update(ec2_conn):
+    """
+    The apt-daily services on the DLAMI cause a conflict upon running any "apt install" commands within the first few
+    minutes of starting an EC2 instance. These services are not necessary for the purpose of the DLC tests, and can
+    therefore be killed. This function kills the services, and then forces "apt-get update" to run in the foreground.
+
+    :param ec2_conn: Fabric SSH connection
+    :return:
+    """
+    apt_daily_services_list = ["apt-daily.service", "apt-daily-upgrade.service", "unattended-upgrades.service"]
+    apt_daily_services = " ".join(apt_daily_services_list)
+    ec2_conn.run(f"sudo systemctl stop {apt_daily_services}")
+    ec2_conn.run(f"sudo systemctl kill --kill-who=all {apt_daily_services}")
+    num_stopped_services = 0
+    for _ in range(60):
```

Comment:
can you explain how you came up with the 60x execution of this loop?

Reply:
It is expected to normally take about 1-2 seconds. The 60 second loop exists to force the execution to wait for a longer amount of time than it would normally take to kill the services.

Also, the loop is meant to be run for 60 seconds, with a sleep of 1 second on each iteration. I missed this while adding the changes. Will fix this.